app/src/main/java/com/mobile/tandil/kotlinbaseproject/fragment/MainDialogFragment.kt

```diff
@@ -0,0 +1,52 @@
+package com.mobile.tandil.kotlinbaseproject.fragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import com.mobile.tandil.kotlinbaseproject.databinding.FragmentMainDialogBinding
+import com.mobile.tandil.kotlinbaseproject.listener.ListenerSetParkingSpaces
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.MainDialogViewModel
+import com.mobile.tandil.kotlinbaseproject.utils.Constants
+
+class MainDialogFragment() : DialogFragment() {
+
+    private lateinit var listener: ListenerSetParkingSpaces
+    private lateinit var binding: FragmentMainDialogBinding
+    private val viewModel = MainDialogViewModel()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        binding = FragmentMainDialogBinding.inflate(inflater)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        listener = arguments?.getSerializable(Constants.LISTENER_PARKING_SPACES_SETTING_KEY) as ListenerSetParkingSpaces
+        setListeners()
+    }
+
+    private fun setListeners() {
+        binding.confirmButton.setOnClickListener {
+            val updateParkingSpaces = binding.parkingAvailableEditText.text.toString()
+            viewModel.updateParkingSpacesValue(updateParkingSpaces)
+            listener.listenerParkingSpaces(updateParkingSpaces)
```

Comment:
this is not following the arch. When the info is saved the viewModel should take the parking size from the model and set a new state on the liveData. Then the activity should react to that state, call this listener and close the dialog

Reply:
the other Mau told me in the comments below to do it in a different way, can we have a call about it?

Reply:
sure, we can talk about it later, just open a chat and will see it